add(epp): applications-tosca-restapi (#2428)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-tosca-restapi.md

---
id: applications-tosca-restapi
title: Tosca Rest API
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Pack assets

### Templates

The Monitoring Connector **Tosca Rest API** brings a host template:

* **App-Tosca-Restapi-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="App-Tosca-Restapi-custom" label="App-Tosca-Restapi-custom">

> | Service Alias | Service Template  | Service Description |
> |:--------------|:------------------|:--------------------|
> 
> > The services listed above are created automatically when the **App-Tosca-Restapi-custom** host template is used.

Reviewer: ```suggestion
By default, no services are created automatically using this template.
```

| Execution-List-Status | App-Tosca-Restapi-Execution-List-Status | Check execution list status |
| Scenario-Status       | App-Tosca-Restapi-Scenario-Status       | Check scenario status       |

> The services listed above are not created automatically when a host template is applied. To use them, [create a service manually](/docs/monitoring/basic-objects/services), then apply the service template you want.

</TabItem>
</Tabs>

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics linked to each service.

<Tabs groupId="sync">
<TabItem value="Execution-List-Status" label="Execution-List-Status">

| Metric name                | Unit  |
|:---------------------------|:------|
| entries.passed.count       | count |
| entries.failed.count       | count |
| entries.not_executed.count | count |
| entries.unknown.count      | count |

</TabItem>
<TabItem value="Scenario-Status" label="Scenario-Status">

| Metric name               | Unit  |
|:--------------------------|:------|
| scenario status           |       |
| scenario.duration.seconds | s     |

</TabItem>
</Tabs>

## Prerequisites

> To monitor, a user with read privileges on the Tosca Commander REST Webservice [API](https://documentation.tricentis.com/devcorner/1000/tcrsapi/) is required.

Reviewer 2: ```suggestion
To monitor this type of resource, a user with read privileges on the Tosca Commander REST Webservice [API](https://documentation.tricentis.com/devcorner/1000/tcrsapi/) is required.
```
